google/api_core/exceptions.py

```diff
@@ -122,12 +124,13 @@ class GoogleAPICallError(GoogleAPIError, metaclass=_GoogleAPICallErrorMeta):
     This may be ``None`` if the exception does not match up to a gRPC error.
     """
 
-    def __init__(self, message, errors=(), details=(), response=None):
+    def __init__(self, message, errors=(), details=(), response=None, error_info=None):
         super(GoogleAPICallError, self).__init__(message)
         self.message = message
         """str: The exception message."""
         self._errors = errors
         self._details = details
+        self._error_info = error_info
```

Comment:
Instead of present entire `error_info`, we would like to add the `reason`, `domain`, and `metadata` fields to the base error exception. go/improving-api-client-errors